VitruvioHost/Plugins/Vitruvio/Source/VitruvioEditor/Private/VitruvioComponentDetails.cpp

```diff
@@ -536,20 +539,19 @@ void AddGenerateButton(IDetailCategoryBuilder& RootCategory, UVitruvioComponent*
 	// clang-format on
 }
 
-void OpenMaterialReplacementDialog(UVitruvioComponent* VitruvioComponent)
+template <typename TInstanceDialogType>
+void OpenReplacementDialog(UVitruvioComponent* VitruvioComponent)
 {
-	UGenerateCompletedCallbackProxy* Proxy = NewObject<UGenerateCompletedCallbackProxy>();
-	Proxy->OnGenerateCompleted.AddLambda([VitruvioComponent]() { FMaterialReplacementDialog::OpenDialog(VitruvioComponent); });
-	VitruvioComponent->Generate(Proxy, {true, false});
+	auto OnDialogClosed = [](const TSharedRef<SWindow>&) {
+		ReplacementDialogOpen = false;
```

Comment:
If the window "crashes" the whole editor will crash (for better or worse ;) ). But otherwise I don't see any way the callback should not be triggered.